src/Connection.php

```diff
@@ -558,19 +556,23 @@ protected function createConnection($hosts, array $config, array $options)
             // force https for port 443
             $scheme = (int) $port === 443 ? 'https' : $scheme;
 
-            return [
-                'host' => $host,
-                'port' => $port,
-                'scheme' => $scheme,
-                'user' => !empty($config['username']) ? $config['username'] : null,
-                'pass' => !empty($config['password']) ? $config['password'] : null,
-            ];
+            $credentials = isset($config['username']) ?
+                $config['username'] . ':' . ($config['password'] ?? '') . '@' :
+                '';
+
+            $fullHost = $scheme . '://' . $credentials . $host . ':' . $port;
+
+            return $fullHost;
         }, $hosts);
 
-        return ClientBuilder::create()
-            ->setHosts($hosts)
-            ->setSelector('\Elasticsearch\ConnectionPool\Selectors\StickyRoundRobinSelector')
-            ->build();
+        $clientBuilder = ClientBuilder::create()
+            ->setHosts($hosts);
+
+        if (isset($options['httpClient'])) {
+            $clientBuilder->setHttpClient($options['httpClient']);
```

Comment:
This is cool! We should be able to pass in the Http fake in the service container and mock all calls to elasticsearch in tests!!!